pkg/filter/filter.go

```diff
@@ -0,0 +1,55 @@
+package filter
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/label"
+	"github.com/giantswarm/microerror"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiutil "sigs.k8s.io/cluster-api/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/giantswarm/azure-admission-controller/pkg/release"
+)
+
+func IsCRProcessed(ctx context.Context, ctrlClient client.Client, objectMeta metav1.Object) (bool, error) {
```

Comment:
CR filtering implementation:
- it first checks for release label
- if previous not found, then it checks GS cluster label (so it can check Cluster's release label)
- if previous not found, then it checks CAPI cluster label (so it can check Cluster's release label)